ui/models/job.js

```diff
@@ -139,6 +139,33 @@ export default class JobModel {
     }
   }
 
+  // Any jobId inside the push will do
+  static async cancelAll(jobId, repoName, ThResultSetStore, thNotify) {
+    const job = await JobModel.get(repoName, jobId);
+    const decisionTaskId = await ThResultSetStore.getGeckoDecisionTaskId(job.result_set_id);
+    const results = await TaskclusterModel.load(decisionTaskId, job);
+    const cancelAllTask = results && results.actions.find(result => result.name === 'cancel-all');
+
+    try {
+      await TaskclusterModel.submit({
+        action: cancelAllTask,
+        decisionTaskId,
+        taskId: results.originalTaskId,
```

Comment:
I think you'll want to omit this.  The cancel-all applies to a whole push, so there's no specific taskId. If no taskId is passed, then there's no need for the extra line in `ui/models/taskcluster.js`.